### PR TITLE
Fix metrics reporting to account for different types of hooks

### PR DESCRIPTION
**Pull request by ddfisher**
Many kinds of hooks can run inside pre-commit, not just pre-commit hooks. We use several of them. Metrics should have distinct namespaces. Now they do.

### CODE REVIEW

File: pre_commit/commands/run.py [hunk 1]
```diff
@@ -322,11 +324,11 @@ def _has_unstaged_config(config_file: str) -> bool:
     return retcode == 1
 
 
-def _run_inner(
+def run(
         config_file: str,
         store: Store,
         args: argparse.Namespace,
-        environ: MutableMapping[str, str],
+        environ: MutableMapping[str, str] = os.environ,
```
Comment (goodspark) [inline pre_commit/commands/run.py:331]: I feel like there might be something iffy here with mutable default values/references, but if this is how pre-commit was coded before, then we don't need to bother tinkering.

**Review by goodspark — APPROVED** [review]
Looks good. If we don't care about parity with existing metrics, then lets merge. If we do, we might want to look into adding the stage name as a tag/metadata instead (which would be new stuff we'd have to add and scope creep a bit).

File: pre_commit/commands/run.py (hunk 1 above)
Comment (ddfisher) [reply pre_commit/commands/run.py:331]: This is how pre-commit was coded before. Generally, doing something like this is a dangerous practice, but here `run` is only ever called once, so it doesn't matter much.

**Comment by ddfisher** [comment]
I think it's okay to not keep parity with existing metrics -- it's only been ~a week.

**Pull request merged**